README.md

```diff
@@ -93,60 +93,60 @@ Instead, computations are performed in a carrier type for the syntax, typically
 
-Different effects make different operations available; see the documentation for individual effects for more information about their operations. Note that we generally don't program against an explicit list of effect components: we take the typeclass-oriented approach, adding new constraints to `sig` as new capabilities become necessary. If you want to name and share some predefined list of effects, it's best to use the `-XConstraintKinds` extension to GHC, capturing the elements of `sig` as a type synonym of kind `Constraint`:
+To add effects to a given computation, add more `Has` constraints to the signature/carrier pair `sig` and `m`. For example, to add a `Reader` effect managing an `Int`, we would write:
 
 ```haskell
-type Shared sig m
-  = ( Has (State String) sig m
-    , Has (Reader Int)   sig m
-    , Has (Writer [String]) sig m
-    )
-
-action3 :: Shared sig m => m ()
-action3 = ask >>= \ i -> put (replicate i '?') >> tell [ "put " ++ show i ++ " '?'s" ]
+action2 :: (Has (State String) sig m, Has (Reader Int) sig m) => m ()
+action2 = do
+  i <- ask
+  put (replicate i '!')
 ```
 
 ### Running effects
 
-Effects are run with _effect handlers_, specified as functions (generally starting with `run…`) unpacking some specific monad with a `Carrier` instance. For example, we can run a `State` computation using `runState`:
+Effects are run with _effect handlers_, specified as functions (generally starting with `run…`) unpacking some specific monad with a `Carrier` instance. For example, we can run a `State` computation using `runState`, imported from the `Control.Carrier.State.Strict` carrier module:
 
 ```haskell
 example1 :: Algebra sig m => [a] -> m (Int, ())
 example1 list = runState 0 $ do
-  i <- get @Int
+  i <- get
   put (i + length list)
 ```
 
-`runState` returns a tuple of both the computed value (the `()`) and the final state (the `Int`), visible in the result of the returned computation. The `get` function is resolved with a visible type application, due to the fact that effects can contain more than one state type (in contrast with `mtl`'s `MonadState`, which limits the user to a single state type).
+`runState` returns a tuple of both the computed value (the `()`) and the final state (the `Int`), visible in the result of the returned computation. The `get` function is resolved with a visible type application, due to the fact that effects can contain more than one state type (in contrast with `mtl`’s `MonadState`, which limits the user to a single state type).
 
 Since this function returns a value in some carrier `m`, effect handlers can be chained to run multiple effects. Here, we get the list to compute the length of from a `Reader` effect:
 
 ```haskell
 example2 :: Algebra sig m => m (Int, ())
 example2 = runReader "hello" . runState 0 $ do
-  list <- ask
-  put (length (list :: String))
+  (list :: String) <- ask
+  put (length list)
```

Comment:
This is a bit nitpicky, but type annotations in patterns require `ScopedTypeVariables`, which is otherwise unnecessary and probably less familiar than writing them in expressions. I think this, and `example3` below, should leave the type annotation on the expression for familiarity’s sake.

Reply:
I think `STV` is clearer here, since the annotation is around the variable rather than its first use, but I don’t feel strongly. Will fix.